tareas/RuizGarcia/Tarea2Diego.jl

```diff
@@ -0,0 +1,642 @@
+
+# ***
+
+# Ahora calculo $\alpha$.
+
+
+
+#-
+function return_min_dist_Sin(vec,val) ## Regresa la distancia mínima al cero de un array. 
+
+    vec =vec.-val
+    @show(vec)
+    vecaux2 = abs.(setdiff!(vec, [0])) ## Elimina el cero del array y convierte todo a valor absoluto.
+    
+    ## Luego, considero dos casos. Esto me permite asegurar que no hay problemas.
+    daux = minimum(vecaux2) ## Calcula el mínimo del array auxiliar.
+    vecaux3 = setdiff!(vecaux2, [daux])
+    
+    if isempty(vecaux3)
+        return daux
+    else
+        d = minimum(vecaux3)
+        return d
+    end
+end
+
+
+
+
+function α_Sin(n)
+    cs=superstable_qs(n)
+    @show(cs)
+    ds=zeros(Float64,n+1) 
+    αs=zeros(Float64,n)
+    
+    ## Primero calcula el valor de las d's
+    for i in 2:length(ds)
+        c=cs[i] ## Valor del parámetro para el cual existe la órbita superestable de periodo 2^(i)
+        vecaux=Float64[xstar for l in 1:2^(n)] ## Vector que guarda la órbias para el mapeo x -> x^2+cs[i]
+        for k in 2:length(vecaux)
+            x=vecaux[k-1]
+            xp=BigFloat(Sⁿc(x,c,1))
+            vecaux[k]=xp
+        end
+        @show(vecaux)
+        ds[i]= return_min_dist_Sin(vecaux,xstar)
+    end
+    @show(ds)
+    
+    for i in 1:n
+        a=-(ds[i]/ds[i+1])
+        αs[i]=a
+    end
+     
+    return αs
+
+end
+
+
+#-
+α_Sin(3)
+
+
+#-
+
+
+
+
+
+
+
+
+
+
+#-
+# ***
+# Por último, hago el diagrama de bifurcaiones del mapeo $S_{c}(x)$.
+
+
+#-
+using Distributions
+using Plots
+
+function bifurcation_Sin(ncs,ns,cfin)
+    cs=range(1,stop=cfin,length=ncs)
+    orbits=zeros(Float64,(ns,ncs))
+    
+    for k in 1:length(cs)
+        c=cs[k]
+        x0=rand(Uniform(0,pi)) ## Genera el punto inicial.
+        aux=zeros(Float64,ns)
+        aux[1]=x0
+        
+        for i in 2:ns
+            xn=aux[i-1]
+            xnp=Sⁿc(xn,c,1)
+            aux[i]=xnp
+        end
+        orbits[:,k]=aux
+    end
+    
+    return (orbits,cs)
+end
+    
+    
+
+function pltOrb_Sin(ncs,ns,cfin)
+    orbits,cs=bifurcation_Sin(ncs,ns,cfin)
+    l=size(orbits)[1]
+    N=Int(floor(l/2))
+    orbitsp=orbits[N:end,:]
+        
+    p=scatter(Float64[cs[1] for j in 1:N],orbitsp[1:N,1],xlabel="c",ylabel="x",label="",color="purple",alpha=0.7,ms=0.9,title="Diagrama de órbitas mapeo f(x)=csin(x)",legend=:topleft)
+    plot!([cs[1],cs[end]],[xstar,xstar],color="red",label="x*",alpha=0.8)
+    plot!([1.5707,1.5707],[0,3],color="blue",label="",alpha=0.6)
+    plot!([2.443,2.443],[0,3],color="blue",label="",alpha=0.6)
+    plot!([2.658,2.658],[0,3],color="blue",label="",alpha=0.6)
+    plot!([2.7063,2.7063],[0,3],color="blue",label="",alpha=0.6)
+    
+    for k in 2:length(cs)
+        scatter!(Float64[cs[k] for j in 1:N],orbitsp[1:N,k],color="purple",alpha=0.7,ms=0.9,label="")
+    end
+    
+    return p 
+end
+
+
+
+#-
+oDiag=pltOrb_Sin(300,500,3*pi)  ## Diagrama de bifurcaciones para el intervalo [1,3π]
```

Comment:
*Indispensable*

En el caso de las gráficas te pido que después hagas coloques un: 
```suggestion
oDiag=pltOrb_Sin(300,500,3*pi)  ## Diagrama de bifurcaciones para el intervalo [1,3π]
savefig("diag_bif1.png")
```
donde diag_bif1.png será el nombre del archivo y después agrega en una celda markdown:

```
![Fig 1](diag_bif1.png "Fig. 1")
```

De está forma te ahorras problemas con la gráfica y el notebook, es una cuestión que afecta principalmente al rendimiento de tu computadora.